libparsec/crates/client/tests/unit/certif/bootstrap_workspace.rs

```diff
@@ -13,7 +16,484 @@ use crate::certif::{CertifBootstrapWorkspaceError, CertificateBasedActionOutcome
+                        .lock()
+                        .unwrap()
+                        .as_ref()
+                        .unwrap()
+                        .clone(),
+                    keys_bundle_access: new_realm_initial_keys_bundle_access
+                        .lock()
+                        .unwrap()
+                        .as_ref()
+                        .unwrap()
+                        .clone(),
+                }
+            }
+        },
+        // 5) Rename
+        {
+            let alice = alice.clone();
+            let new_realm_certificates = new_realm_certificates.clone();
+            move |req: authenticated_cmds::latest::realm_rename::Req| {
+                RealmNameCertificate::verify_and_load(
+                    &req.realm_name_certificate,
+                    &alice.verify_key(),
+                    &alice.device_id,
+                    Some(realm_id),
+                )
+                .unwrap();
+                new_realm_certificates
+                    .lock()
+                    .unwrap()
+                    .push(req.realm_name_certificate);
+                authenticated_cmds::latest::realm_rename::Rep::Ok
+            }
+        }
+    );
+
+    let res = ops
+        .bootstrap_workspace(realm_id, &"wksp1".parse().unwrap())
+        .await
+        .unwrap();
+
+    p_assert_matches!(res, CertificateBasedActionOutcome::Uploaded { .. });
+}
+
+#[parsec_test(testbed = "minimal")]
+async fn ok_partial_bootstrap_realm_created(env: &TestbedEnv) {
+    let (env, realm_id) = env.customize_with_map(|builder| {
+        builder.certificates_storage_fetch_certificates("alice@dev1");
+        builder.new_realm("alice").map(|event| event.realm_id)
+    });
+    let alice = env.local_device("alice@dev1");
+    let ops = certificates_ops_factory(&env, &alice).await;
+
+    let new_realm_certificates: Arc<Mutex<Vec<Bytes>>> = Arc::default();
+    let new_realm_initial_keys_bundle: Arc<Mutex<Option<Bytes>>> = Arc::default();
+    let new_realm_initial_keys_bundle_access: Arc<Mutex<Option<Bytes>>> = Arc::default();
+    test_register_sequence_of_send_hooks!(
+        &env.discriminant_dir,
+        // 1) Realm creation
+        {
+            let alice = alice.clone();
+            let new_realm_certificates = new_realm_certificates.clone();
+            move |req: authenticated_cmds::latest::realm_create::Req| {
+                RealmRoleCertificate::verify_and_load(
+                    &req.realm_role_certificate,
+                    &alice.verify_key(),
+                    CertificateSignerRef::User(&alice.device_id),
+                    Some(realm_id),
+                    Some(alice.user_id()),
+                )
+                .unwrap();
+                new_realm_certificates
+                    .lock()
+                    .unwrap()
+                    .push(req.realm_role_certificate);
+                authenticated_cmds::latest::realm_create::Rep::Ok
```

Comment:
Realm already exists in this test, so the result shouldn't be `ok` but `realm_already_exists` !

Reply:
Ah, that's what I missed !

Reply:
Done